components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationRequestBuilder.java

```diff
@@ -0,0 +1,158 @@
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.action.execution.ActionExecutionRequestBuilder;
+import org.wso2.carbon.identity.action.execution.exception.ActionExecutionRequestBuilderException;
+import org.wso2.carbon.identity.action.execution.model.ActionExecutionRequest;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.execution.model.Event;
+import org.wso2.carbon.identity.action.execution.model.Request;
+import org.wso2.carbon.identity.action.execution.model.Tenant;
+import org.wso2.carbon.identity.action.execution.model.UserStore;
+import org.wso2.carbon.identity.action.execution.model.User;
+import org.wso2.carbon.identity.action.execution.model.Organization;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthHistory;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.UserIdNotFoundException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.adapter.internal.AuthenticatorAdapterDataHolder;
+import org.wso2.carbon.identity.application.authenticator.adapter.model.AllowedOperationBuilder;
+import org.wso2.carbon.identity.application.authenticator.adapter.model.AuthenticationRequestEvent;
+import org.wso2.carbon.identity.application.authenticator.adapter.model.AuthenticationRequestEvent.AuthenticatedStep;
+import org.wso2.carbon.identity.application.authenticator.adapter.model.AuthenticationRequestUser;
+import org.wso2.carbon.identity.application.authenticator.adapter.model.AuthenticationRequest;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.Map;
+
+/**
+ * This is a builder class which is responsible for building authentication request payload which will be sent to the
+ * external authentication service.
+ */
+public class AuthenticationRequestBuilder implements ActionExecutionRequestBuilder {
+
+    private static final Log LOG = LogFactory.getLog(AuthenticationRequestBuilder.class);
+
+    @Override
+    public ActionType getSupportedActionType() {
+
+        return ActionType.AUTHENTICATION;
+    }
+
+    @Override
+    public ActionExecutionRequest buildActionExecutionRequest(Map<String, Object> eventContext)
+            throws ActionExecutionRequestBuilderException {
+
+        HttpServletRequest request = (HttpServletRequest) eventContext.get(AuthenticatorAdapterConstants.AUTH_REQUEST);
+        AuthenticationContext context = (AuthenticationContext) eventContext.get(
```

Comment:
Doesn't authenticationContext includes the ServletRequest

Reply:
No it only contains the initial request. The current servlet request.